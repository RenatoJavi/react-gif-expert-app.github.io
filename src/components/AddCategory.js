import React, { useState } from 'react';

import PropTypes from 'prop-types';

export const AddCategory  = ({setCategories}) => {



const [inputValue, setInputValue] = useState("Hola soy programador");
const handletInputChange=(e)=>{
    
setInputValue(e.target.value);

};
const handleSubmit=(evento)=>{
    evento.preventDefault();//=>para prevenir el comportamineto por defecto del formulario
    console.log('Submit al 100%');
    if (inputValue.trim().length>2) {

        setCategories(cats=>[inputValue,...cats]);
        setInputValue('');
    } else {
        console.warn('error en la ejecucion de cargar datos' );
    }
   
    
}

    return (       
           <form onSubmit={handleSubmit}>
           <input 
            type="text" 
            value={inputValue}
           onChange={handletInputChange}          
            
            ></input>
           </form>
         
       
    )
}

AddCategory.prototype={
setCategories:PropTypes.func.isRequired

}