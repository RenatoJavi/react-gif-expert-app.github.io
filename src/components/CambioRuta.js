import React from 'react'

export const CambioRuta = () => {
    return (
        <div>
            
        </div>
    )
}
