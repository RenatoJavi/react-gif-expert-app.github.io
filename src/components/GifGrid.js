
import React from 'react'
import { useFechGifs } from '../hooks/useFechGifs'
import { GifGridItem } from './GifGridItem';


export const GifGrid = ({category}) => {
  const {data:images,loading}=  useFechGifs(category);

 
//const [contador, setcontador] = useState(0);
// useEffect(()=>{
//     getGifs(category)
//     .then(imgs=>{setImages(imgs)});
// },[category]);

    // const getGifs=async()=>{

    //  const url=`https://api.giphy.com/v1/gifs/search?q=${encodeURI(category)}&limit=10&api_key=k3yngY6Tvh8ivUgBnE8H4cPWpsn6xLV7`;
    //  const resp=await  fetch(url);
    //  const {data}=await resp.json();

    //  const gifs=data.map(img=>{
    //     return{
    //         id:img.id,
    //         title:img.title,
    //         url:img.images?.downsized_medium.url          
    //     }
    //  });
    //  console.log(gifs);
    //  setImages(gifs);

    //  //setImagenes
    // };
    //getGifs();
    return (//aqui estoy regresando 2 objetos. por eso uso el fracment
       <>
        <h3>{category}</h3>
        {loading && <p>Loading ...</p>}
             <div className="card-grid" >
                        
               {
                images.map(img=>(
                    
                    <GifGridItem 
                        key={img.id}
                    { ...img}/>
                ))
               }
                     
        </div>
         {/*<h3>{contador}</h3>
             <button onClick={()=>{setcontador(contador+1)}}>Add</button> */}
        </>
    )
}
