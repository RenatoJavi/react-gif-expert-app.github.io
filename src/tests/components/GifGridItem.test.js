import React from 'react';
import {shallow} from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';


describe('Pruebas en <GitGridItem/>', () => {
    const title='Un titulo';
    const url='http://localhost/algo.jpg';

    test('should de mostrar el texto correctamente ', () => {      
        const wrapper= shallow(<GifGridItem title={title} url={url}/>)
        expect(wrapper).toMatchSnapshot();
    })
    
});
