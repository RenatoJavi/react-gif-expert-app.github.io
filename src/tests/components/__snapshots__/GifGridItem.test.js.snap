// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GitGridItem/> should de mostrar el texto correctamente  1`] = `
<div
  className="card animate__animated animate__rubberBand animate__delay-4s "
>
  <img
    alt="Un titulo"
    src="http://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
